src/main/cpp/blaze_util_windows.cc

```diff
@@ -406,6 +406,12 @@ string GetOutputRoot() {
 }
 
 string GetHomeDir() {
+  if (IsRunningWithinTest()) {
+    // Bazel is running inside of a test. Respect $HOME that the test setup has
+    // set instead of using the actual home directory of the current user.
+    return GetEnv("HOME");
```

Comment:
nit: not that this is very important, but can you maybe get away with `$TEST_TMPDIR` instead? It's kinda confusing because `$HOME` usually means, well, the home directory of some user.

Reply:
On Linux we do the same: https://github.com/bazelbuild/bazel/blob/420b9533534951b82c1e13e6d83fa67390e95778/src/main/cpp/blaze_util_posix.cc#L205

Bazel actually sets a mock $HOME for the test.